Source/WebCore/platform/ScrollSnapAnimatorState.cpp

```diff
@@ -151,23 +183,19 @@ bool ScrollSnapAnimatorState::resnapAfterLayout(ScrollOffset scrollOffset, const
         activeHorizontalIndex = activeSnapIndexForAxis(ScrollEventAxis::Horizontal);
     }
     
-    // If we have active targets, see if we need to preserve one of them
-    if (activeHorizontalID && activeHorizontalIndex && activeVerticalID && activeVerticalIndex && *activeHorizontalID
-        != snapOffsetsHorizontal[*activeHorizontalIndex].snapTargetID && *activeVerticalID
-        != snapOffsetsVertical[*activeVerticalIndex].snapTargetID)
-        snapPointChanged |= preserveCurrentTargetForAxis(ScrollEventAxis::Horizontal);
-
-    // If we do not have current targets and are snapped to multiple targets, set them
-    if ((!activeHorizontalID && activeHorizontalIndex) && (!activeVerticalID && activeVerticalIndex)) {
-        auto horizontalID = snapOffsetsForAxis(ScrollEventAxis::Horizontal)[*activeHorizontalIndex].snapTargetID;
-        auto verticalID = snapOffsetsForAxis(ScrollEventAxis::Vertical)[*activeVerticalIndex].snapTargetID;
-        if (horizontalID && verticalID && horizontalID != verticalID) {
-            setActiveSnapIndexIDForAxis(ScrollEventAxis::Horizontal, horizontalID);
-            setActiveSnapIndexIDForAxis(ScrollEventAxis::Vertical, verticalID);
-        }
-    }
+    auto verticalOffsets = currentlySnappedOffsetsForAxis(ScrollEventAxis::Vertical);
+    auto horizontalOffsets = currentlySnappedOffsetsForAxis(ScrollEventAxis::Horizontal);
     
-    LOG_WITH_STREAM(ScrollSnap, stream << "ScrollSnapAnimatorState::resnapAfterLayout() current target: horizontal: " << activeHorizontalID << " vertical: "<< activeVerticalID);
+    if (isSnappedToMultipleBoxes(m_currentlySnappedBoxesX, m_currentlySnappedBoxesY) && !isSnappedToMultipleBoxes(verticalOffsets, horizontalOffsets)) {
```

Comment:
You should keep the logging and have it indicate what the result of resnapping was.

Reply:
Fixed.